src/e2e-test/java/io/cdap/cdap/ui/utils/Helper.java

```diff
@@ -217,4 +271,42 @@ public static void setCdapTheme(String themePath) {
       Assert.fail(e.getMessage());
     }
   }
+
+  public static void takeScreenShot() {
+    StackTraceElement[] stacktrace = Thread.currentThread().getStackTrace();
+    StackTraceElement e = stacktrace[2];
+    takeScreenShot(e.getClassName(), e.getMethodName(), "");
+  }
+
+  public static void takeScreenShot(String postfix) {
+    StackTraceElement[] stacktrace = Thread.currentThread().getStackTrace();
+    StackTraceElement e = stacktrace[2];
+    takeScreenShot(e.getClassName(), e.getMethodName(), postfix);
+  }
+
+  public static void takeScreenShot(String className, String methodName, String postfix) {
+    try {
+      TakesScreenshot scrShot = (TakesScreenshot) SeleniumDriver.getDriver();
+      File srcFile = scrShot.getScreenshotAs(OutputType.FILE);
+      File destFile = new File(String.format("/tmp/cdap-ui-integration-fixtures/selenium/snapshots/%s/%s_%s.png",
+                                             className,
+                                             methodName,
+                                             postfix));
+      FileUtils.copyFile(srcFile, destFile);
+    } catch (IOException e) {
+      logger.info("cannot take a screenshot");
+    }
+  }
+
+  public static void waitSeconds() {
+    waitSeconds(1);
+  }
+
+  public static void waitSeconds(long second) {
```

Comment:
why do we need this? we better not use `Thread.sleep` .

Reply:
There are some loading happening

Reply:
Can't you wait for an element to appear (indicating the wait is over)?

Reply:
I will change

Reply:
actually some of the waits are designed to wait for new data refresh. These cases do not have new elements to wait for

Reply:
Agree that we don't wait arbitrary time, instead wait for a target element to show up